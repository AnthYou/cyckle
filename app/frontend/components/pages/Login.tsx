import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

import Button from "../UI/Button";
import ManDoor from "@/images/login.svg";
import Input from "../UI/Input";
import { Credentials, loginUser } from '@/store/actions/auth';
import { useAppDispatch, useAppSelector } from '@/hooks/redux';
import { messageActions } from '@/store/slices/message-slice';

const defaultValues: Credentials = {
  email: "",
  password: ""
};

const Login = () => {
  const dispatch = useAppDispatch();
  const { message: error } = useAppSelector(state => state.message);
  const [formValues, setFormValues] = useState<Credentials>(defaultValues);
  const isValid = formValues.email.includes("@") && formValues.password.length >= 6;

  useEffect(() => {
    dispatch(messageActions.clearMessage());

    return () => { dispatch(messageActions.clearMessage()) };
  }, [dispatch])

  const handleFormSubmit = (event: React.FormEvent) => {
    event.preventDefault();

    dispatch(loginUser(formValues));
  };

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;

    setFormValues((prevValues) => {
      return {
        ...prevValues,
        [name]: value,
      };
    });
  }

  return (
    <>
      <h1>Connectez-vous</h1>
      <div className="flex flex-col sm:flex-row items-center justify-between gap-4">
        <form onSubmit={handleFormSubmit} className="w-full sm:w-1/2">
          <Input
            id="email"
            type="text"
            label="Email"
            name="email"
            value={formValues.email}
            onChange={handleInputChange}
            placeholder="raymond@cyckle.com"
            required={true}
          />
          <Input
            id="password"
            type="password"
            label="Mot de passe"
            name="password"
            placeholder="******"
            value={formValues.password}
            onChange={handleInputChange}
            required={true}
          />
          {error && <p className="text-red-500">{error}</p>}
          <div className="flex flex-col items-center justify-center gap-4">
            <Button color="primary" className="mt-5" disabled={!isValid}>
              S'inscrire
            </Button>
            <p>
              Vous n'avez pas encore de compte ? <Link to="/signup">Inscrivez-vous</Link>
            </p>
          </div>
        </form>
        <img
          src={ManDoor}
          alt="A man opening a door."
          className="w-full sm:w-1/2"
        />
      </div>
    </>
  );
}

export default Login;
