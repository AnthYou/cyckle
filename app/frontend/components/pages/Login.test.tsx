import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import Login from "./Login";

describe("<Login />", () => {
  it("should render correctly", () => {
    const component = renderer.create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
