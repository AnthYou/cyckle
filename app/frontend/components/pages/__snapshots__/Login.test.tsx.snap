// Vitest Snapshot v1

exports[`<Login /> > should render correctly 1`] = `
[
  <h1>
    Connectez-vous
  </h1>,
  <div
    className="flex flex-col sm:flex-row items-center justify-between gap-4"
  >
    <form
      className="w-full sm:w-1/2"
      onSubmit={[Function]}
    >
      <div
        className="_Input_658f0a"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="raymond@cyckle.com"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="_Input_658f0a"
      >
        <label
          htmlFor="password"
        >
          Mot de passe
        </label>
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="******"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="flex flex-col items-center justify-center gap-4"
      >
        <button
          className="_Button_4ea4ec _primary_4ea4ec mt-5"
          disabled={true}
        >
          S'inscrire
        </button>
        <p>
          Vous n'avez pas encore de compte ? 
          <a
            href="/signup"
            onClick={[Function]}
          >
            Inscrivez-vous
          </a>
        </p>
      </div>
    </form>
    <img
      alt="A man opening a door."
      className="w-full sm:w-1/2"
      src="/@fs/@/images/login.svg"
    />
  </div>,
]
`;
